<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>IsplCodeGen</class>
 <widget class="QMainWindow" name="IsplCodeGen">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>896</width>
    <height>755</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>IsplCodeGen</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QTabWidget" name="tabWidget">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>0</y>
      <width>581</width>
      <height>661</height>
     </rect>
    </property>
    <property name="currentIndex">
     <number>0</number>
    </property>
    <widget class="QWidget" name="tab">
     <attribute name="title">
      <string>Decison variable</string>
     </attribute>
     <widget class="QWidget" name="widget" native="true">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>0</y>
        <width>1071</width>
        <height>962</height>
       </rect>
      </property>
      <widget class="QWidget" name="widget_2" native="true">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>160</y>
         <width>561</width>
         <height>121</height>
        </rect>
       </property>
       <layout class="QHBoxLayout" name="horizontalLayout">
        <item>
         <widget class="QLabel" name="VarsLabel">
          <property name="text">
           <string>Var    </string>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="horizontalSpacer">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeType">
           <enum>QSizePolicy::Fixed</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>18</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
        <item>
         <widget class="QTextEdit" name="Var_plainTextEdit"/>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="widget_3" native="true">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>280</y>
         <width>561</width>
         <height>121</height>
        </rect>
       </property>
       <layout class="QHBoxLayout" name="horizontalLayout_2">
        <item>
         <widget class="QLabel" name="label_2">
          <property name="text">
           <string>Action</string>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="horizontalSpacer_2">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeType">
           <enum>QSizePolicy::Fixed</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
        <item>
         <widget class="QTextEdit" name="action_plainTextEdit"/>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="widget_4" native="true">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>400</y>
         <width>561</width>
         <height>121</height>
        </rect>
       </property>
       <layout class="QHBoxLayout" name="horizontalLayout_3">
        <item>
         <widget class="QLabel" name="label_3">
          <property name="text">
           <string>Protocol</string>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="horizontalSpacer_3">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeType">
           <enum>QSizePolicy::Fixed</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>5</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
        <item>
         <widget class="QTextEdit" name="Protocol_plainTextEdit"/>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="widget_5" native="true">
       <property name="geometry">
        <rect>
         <x>9</x>
         <y>743</y>
         <width>298</width>
         <height>210</height>
        </rect>
       </property>
       <layout class="QHBoxLayout" name="horizontalLayout_4">
        <item>
         <widget class="QLabel" name="label_4">
          <property name="text">
           <string>Var</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QTextEdit" name="textEdit_5"/>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="widget_6" native="true">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>50</y>
         <width>561</width>
         <height>41</height>
        </rect>
       </property>
       <layout class="QHBoxLayout" name="horizontalLayout_5">
        <item>
         <widget class="QLabel" name="AgentName_label">
          <property name="text">
           <string>Decision variable nameï¼š</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLineEdit" name="AgentName_lineEdit"/>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="widget_8" native="true">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>520</y>
         <width>561</width>
         <height>111</height>
        </rect>
       </property>
       <layout class="QHBoxLayout" name="horizontalLayout_7">
        <item>
         <widget class="QLabel" name="label_7">
          <property name="text">
           <string>Evolution</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QTextEdit" name="Evolution_plainTextEdit"/>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="widget_7" native="true">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>10</y>
         <width>551</width>
         <height>41</height>
        </rect>
       </property>
       <layout class="QHBoxLayout" name="horizontalLayout_6">
        <item>
         <widget class="QLabel" name="AgentMode_label">
          <property name="text">
           <string>Agent Mode:</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QComboBox" name="AgentMode_comboBox">
          <item>
           <property name="text">
            <string>Environment</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Normal Agent</string>
           </property>
          </item>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="widget_11" native="true">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>100</y>
         <width>561</width>
         <height>41</height>
        </rect>
       </property>
       <layout class="QHBoxLayout" name="horizontalLayout_10">
        <item>
         <widget class="QLabel" name="AgentName_label_2">
          <property name="text">
           <string>Sensing varible num</string>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="horizontalSpacer_4">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeType">
           <enum>QSizePolicy::Fixed</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>25</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
        <item>
         <widget class="QLineEdit" name="SensingVarNum_lineEdit"/>
        </item>
       </layout>
      </widget>
     </widget>
    </widget>
    <widget class="QWidget" name="tab_2">
     <attribute name="title">
      <string>Evaluation</string>
     </attribute>
     <widget class="QWidget" name="widget_9" native="true">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>10</y>
        <width>571</width>
        <height>121</height>
       </rect>
      </property>
      <layout class="QHBoxLayout" name="horizontalLayout_8">
       <item>
        <widget class="QLabel" name="label_8">
         <property name="text">
          <string>Evolution</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QTextEdit" name="Evaluation_plaintext"/>
       </item>
       <item>
        <widget class="QPushButton" name="EvaluationCodeGenBtn">
         <property name="text">
          <string>generate code</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
    <widget class="QWidget" name="tab_3">
     <attribute name="title">
      <string>InitStates AND Group</string>
     </attribute>
     <widget class="QWidget" name="widget_10" native="true">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>10</y>
        <width>561</width>
        <height>121</height>
       </rect>
      </property>
      <layout class="QHBoxLayout" name="horizontalLayout_9">
       <item>
        <widget class="QLabel" name="label_9">
         <property name="text">
          <string>Evolution</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QTextEdit" name="InitStates_plainText"/>
       </item>
       <item>
        <widget class="QPushButton" name="InitStatesCodeGenBtn">
         <property name="text">
          <string>generate code</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="widget_21" native="true">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>140</y>
        <width>561</width>
        <height>121</height>
       </rect>
      </property>
      <layout class="QHBoxLayout" name="horizontalLayout_19">
       <item>
        <widget class="QLabel" name="label_19">
         <property name="text">
          <string>Group</string>
         </property>
        </widget>
       </item>
       <item>
        <spacer name="horizontalSpacer_7">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="sizeType">
          <enum>QSizePolicy::Fixed</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>15</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
       <item>
        <widget class="QTextEdit" name="Groups_plainText"/>
       </item>
       <item>
        <widget class="QPushButton" name="GroupsCodeGenBtn">
         <property name="text">
          <string>generate code</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
    <widget class="QWidget" name="tab_4">
     <attribute name="title">
      <string>Formulae</string>
     </attribute>
     <widget class="QWidget" name="widget_22" native="true">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>0</y>
        <width>571</width>
        <height>121</height>
       </rect>
      </property>
      <layout class="QHBoxLayout" name="horizontalLayout_20">
       <item>
        <widget class="QLabel" name="label_20">
         <property name="text">
          <string>Formulate</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QTextEdit" name="Formulae_plaintext"/>
       </item>
       <item>
        <widget class="QPushButton" name="FormulateCodeGenBtn">
         <property name="text">
          <string>generate code</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
   </widget>
   <widget class="QTextEdit" name="plainTextEditPreview">
    <property name="geometry">
     <rect>
      <x>580</x>
      <y>20</y>
      <width>291</width>
      <height>641</height>
     </rect>
    </property>
   </widget>
   <widget class="QPushButton" name="AgentCodeGenBtn">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>670</y>
      <width>191</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>GENERATE AGENT TO MODEL</string>
    </property>
   </widget>
   <widget class="QPushButton" name="copyCurAgentBtn">
    <property name="geometry">
     <rect>
      <x>240</x>
      <y>670</y>
      <width>181</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>SAVE AS TEMPLATE</string>
    </property>
   </widget>
   <widget class="QPushButton" name="clearIsplCodeBtn">
    <property name="geometry">
     <rect>
      <x>640</x>
      <y>670</y>
      <width>221</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>CLEAR CODE</string>
    </property>
   </widget>
   <widget class="QPushButton" name="GenWithTemplateBtn">
    <property name="geometry">
     <rect>
      <x>430</x>
      <y>670</y>
      <width>191</width>
      <height>31</height>
     </rect>
    </property>
    <property name="text">
     <string>GENERATE WITH TEMPLATE</string>
    </property>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>896</width>
     <height>23</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionNew"/>
    <addaction name="separator"/>
    <addaction name="actionExport"/>
   </widget>
   <addaction name="menuFile"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <action name="actionNew">
   <property name="text">
    <string> New</string>
   </property>
  </action>
  <action name="actionExport">
   <property name="text">
    <string>Export</string>
   </property>
  </action>
 </widget>
 <resources/>
 <connections/>
</ui>
